go_covert_lib/client/src/ui-components/NumberInput.jsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import FormControl from 'react-bootstrap/FormControl';
+import InputGroup from 'react-bootstrap/InputGroup';
+
+const NumberInput = (props) => {
+  const { label, value, onChange } = props;
+  return (
+    <InputGroup className="m-1 w-25">
+      <InputGroup.Prepend>
+        <InputGroup.Text className="input-text">{label}</InputGroup.Text>
+      </InputGroup.Prepend>
+      <FormControl
+        value={value}
+        onChange={onChange}
+      />
+    </InputGroup>
+  );
+};
+
+NumberInput.propTypes = {
+  label: PropTypes.string.isRequired,
```

Comment:
The number ParamType comes with a range. This range should be used to limit the values that are provided.
- The min and max values of the number input must be set.
- The onChange method should check that the user has not manually entered a value that is too large.

Reply:
Ah true, yeah I'll add input validation for them